File: frontend/angular-app/src/app/models/communication-log.ts
export type CommunicationType = 'Vocabulary' | 'Speaking' | 'Conversation' | 'Interview' | 'Writing';

export interface CommunicationLog {
  id: number;
  type: CommunicationType;
  content: string;
  confidenceLevel: number;
  notes?: string | null;
  createdAt: string;
}

export interface CommunicationTodayPlan {
  date: string;
  word: string;
  meaning: string;
  exampleSentence: string;
  speakingPrompt: string;
  confidenceDrill: string;
}

File: backend/Jarvis.Api/Models/CommunicationLog.cs
namespace Jarvis.Api.Models;

public class CommunicationLog
{
    public int Id { get; set; }

    public string Type { get; set; } = string.Empty;

    public string Content { get; set; } = string.Empty;

    public int ConfidenceLevel { get; set; }

    public string? Notes { get; set; }

    public DateTime CreatedAt { get; set; }
}
